gocd/values.yaml

```diff
@@ -255,27 +255,33 @@ server:
       defaultMode:
 
 agent:
+  # agent.enabled is the toggle to deploy static GoCD Agents. Change to false for Server Only Deployment.
+  enabled: true
+
   # specifies overrides for agent specific service account creation
   serviceAccount:
     # specifies whether the top level service account (also used by the server) should be reused as the service account for gocd agents
     reuseTopLevelServiceAccount: false
     # if reuseTopLevelServiceAccount is false, this field specifies the name of an existing service account to be associated with gocd agents
     # If field is empty, the service account "default" will be used.
     name:
+    # agent.serviceAccount.automountServiceAccountToken specifies whether to automount the service account token in the GoCD Agent pods
     automountServiceAccountToken: false
 
-  # agent.deployment.labels is the labels for the GoCD Agent Deployment
-  deployment:
+  controller:
+    # agent.controller.kind is the type of controller used to deploy Agents. Use either Deployment (default), StatefulSet or DaemonSet
+    kind: "Deployment"
+    # agent.controller.labels is the labels for the GoCD Agent controller
     labels: {}
-  # agent.pod.labels is the labels for the GoCD Agent Pods
+    # agent.controller.annotations is the annotations for the GoCD Agent controller
+    annotations:
+      # iam.amazonaws.com/role: arn:aws:iam::xxx:role/my-custom-role
+
   pod:
+    # agent.pod.labels is the labels for the GoCD Agent Pods
     labels: {}
-
-  # agent.annotations is the annotations for the GoCD Agent Deployment and Pod Spec
-  annotations:
-    deployment:
-      # iam.amazonaws.com/role: arn:aws:iam::xxx:role/my-custom-role
-    pod:
+    # agent.pod.annotations is the annotations for the GoCD Agent Pods
+    annotations:
```

Comment:
I think this is all unnecessary breaking change for existing users. While it's "purer", there's no compelling need to back breakward compatibility for existing users.

Suggest

* Add `agent.kind` for `Deployment` vs `StatefulSet`
* Leave the existing structure named the way they are, but apply to both deployment and statefulset, despite the naming. Change the comment to make this clear it's applied regardless of the `kind`

Reply:
Fair, this is what I meant by
>  There are many other ways to write those keys, I await the maintainers' opinion.

There is a good amount of incoherence across `server` & `agent` and `annotations` & `labels`, but I get you're not trying to make 3.0.0 happen on this PR.

10m of work, will do